`azd init` select template panics when using default selection

1. Create empty directory
2. Run `azd init`
3. Select "Select a template"
4. Press enter. This selects the default selection of `Minimal`

Expected:
Prompt should succeed

Actual:

```
panic: runtime error: index out of range [-2]

goroutine 1 [running]:
github.com/azure/azure-dev/cli/azd/pkg/templates.PromptTemplate({0x562ddd691948, 0xc0005f25a0}, {0x562ddce922fd, 0x1a}, 0x0?, {0x562ddd69eec8, 0xc00034c750})
        /mnt/vss/_work/1/s/cli/azd/pkg/templates/template_manager.go:229 +0xc5e
github.com/azure/azure-dev/cli/azd/cmd.(*initAction).initializeTemplate(0xc00011b020, {0x562ddd691948, 0xc0005f25a0}, 0x2?)
        /mnt/vss/_work/1/s/cli/azd/cmd/init.go:299 +0x76
github.com/azure/azure-dev/cli/azd/cmd.(*initAction).Run(0xc00011b020, {0x562ddd691948, 0xc0005f25a0})
        /mnt/vss/_work/1/s/cli/azd/cmd/init.go:199 +0x6de
```

--- a/cli/azd/pkg/input/console.go
+++ b/cli/azd/pkg/input/console.go
@@ -708,6 +708,10 @@ func (c *AskerConsole) Select(ctx context.Context, options ConsoleOptions) (int,
 	}
 
 	surveyOptions := make([]string, len(options.Options))
+	surveyDefault := options.DefaultValue
+	surveyDefaultAsString, surveyDefaultIsString := surveyDefault.(string)
+
+	// Modify the options and default value to include any details
 	for i, option := range options.Options {
 		surveyOptions[i] = option
 
@@ -718,13 +722,17 @@ func (c *AskerConsole) Select(ctx context.Context, options ConsoleOptions) (int,
 			} else {
 				surveyOptions[i] += "\n"
 			}
+
+			if surveyDefaultIsString && surveyDefaultAsString == option {
+				surveyDefault = surveyOptions[i]
+			}
 		}
 	}
 
 	survey := &survey.Select{
 		Message: options.Message,
 		Options: surveyOptions,
-		Default: options.DefaultValue,
+		Default: surveyDefault,
 		Help:    options.Help,
 	}
 
@@ -773,6 +781,9 @@ func (c *AskerConsole) MultiSelect(ctx context.Context, options ConsoleOptions) 
 	}
 
 	surveyOptions := make([]string, len(options.Options))
+	surveyDefault := options.DefaultValue
+	surveyDefaultAsArr, surveyDefaultIsArr := surveyDefault.([]string)
+	// Modify the options and default value to include any details
 	for i, option := range options.Options {
 		surveyOptions[i] = option
 
@@ -780,12 +791,20 @@ func (c *AskerConsole) MultiSelect(ctx context.Context, options ConsoleOptions) 
 			detailString := output.WithGrayFormat("%s", options.OptionDetails[i])
 			surveyOptions[i] += fmt.Sprintf("\n  %s\n", detailString)
 		}
+
+		if surveyDefaultIsArr {
+			for idx, defaultOption := range surveyDefaultAsArr {
+				if defaultOption == option {
+					surveyDefaultAsArr[idx] = surveyOptions[i]
+				}
+			}
+		}
 	}
 
 	survey := &survey.MultiSelect{
 		Message: options.Message,
 		Options: surveyOptions,
-		Default: options.DefaultValue,
+		Default: surveyDefault,
 		Help:    options.Help,
 	}
 

